SU2_DEF/src/drivers/CDiscAdjDeformationDriver.cpp

```diff
@@ -0,0 +1,945 @@
+
+    StartTime = SU2_MPI::Wtime();
+
+    /*--- Preprocessing of the geometry for all zones. ---*/
+
+    Geometrical_Preprocessing();
+
+    /*--- Preprocessing of the outputs for all zones. ---*/
+
+    Output_Preprocessing();
+
+    /*--- Preprocessing time is reported now, but not included in the next compute portion. ---*/
+
+    StopTime = SU2_MPI::Wtime();
+
+    /*--- Compute/print the total time for performance benchmarking. ---*/
+
+    UsedTime = StopTime-StartTime;
+    UsedTimePreproc    = UsedTime;
+    UsedTimeCompute    = 0.0;
+
+}
+
+CDiscAdjDeformationDriver::~CDiscAdjDeformationDriver(void) {
+
+}
+
+void CDiscAdjDeformationDriver::SetContainers_Null() {
+
+    /*--- Definition of the containers per zones ---*/
+
+    config_container    = new CConfig*[nZone] ();
+    geometry_container  = new CGeometry**[nZone] ();
+    surface_movement    = new CSurfaceMovement*[nZone] ();
+    grid_movement       = new CVolumetricMovement*[nZone] ();
+
+    nInst               = new unsigned short[nZone];
+    for (iZone = 0; iZone < nZone; iZone++) {
+        nInst[iZone] = 1;
+    }
+}
+
+void CDiscAdjDeformationDriver::Input_Preprocessing() {
+
+    // TODO Check that config is for Discrete Adjoint !
+
+    /*--- Initialize a char to store the zone filename ---*/
+    char zone_file_name[MAX_STRING_SIZE];
+
+    /*--- Loop over all zones to initialize the various classes. In most
+      cases, nZone is equal to one. This represents the solution of a partial
+      differential equation on a single block, unstructured mesh. ---*/
+
+    for (iZone = 0; iZone < nZone; iZone++) {
+
+         /*--- Definition of the configuration option class for all zones. In this
+          constructor, the input configuration file is parsed and all options are
+          read and stored. ---*/
+
+        if (driver_config->GetnConfigFiles() > 0){
+            strcpy(zone_file_name, driver_config->GetConfigFilename(iZone).c_str());
+            config_container[iZone] = new CConfig(driver_config, zone_file_name, SU2_COMPONENT::SU2_DOT, iZone, nZone, true);
+        } else {
+            config_container[iZone] = new CConfig(driver_config, config_file_name, SU2_COMPONENT::SU2_DOT, iZone, nZone, true);
+        }
+
+        config_container[iZone]->SetMPICommunicator(SU2_MPI::GetComm());
+    }
+
+    /*--- Set the multizone part of the problem. ---*/
+    if (driver_config->GetMultizone_Problem()){
+        for (iZone = 0; iZone < nZone; iZone++) {
+            /*--- Set the interface markers for multizone ---*/
+            config_container[iZone]->SetMultizone(driver_config, config_container);
+        }
+    }
+}
+
+void CDiscAdjDeformationDriver::Geometrical_Preprocessing() {
+
+    /*--- Loop over all zones to initialize the various classes. In most
+     cases, nZone is equal to one. This represents the solution of a partial
+     differential equation on a single block, unstructured mesh. ---*/
+
+     for (iZone = 0; iZone < nZone; iZone++) {
```

Comment:
If possible it would be good to share methods like these between the two deformation drivers, for example DiscAdj could inherit from the normal, or the two could share a base, etc.